deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -19,25 +50,61 @@ GrB_Info GxB_MatrixTupleIter_new
 	GxB_MatrixTupleIter **iter,     // iterator to create
 	GrB_Matrix A                    // matrix to iterate over
 ) {
-	GB_WHERE(A, "GxB_MatrixTupleIter_new (A)") ;
+	GB_WHERE(A, "GxB_MatrixTupleIter_iterate_row (A)") ;
 	GB_RETURN_IF_NULL_OR_FAULTY(A) ;
 
-	// make sure matrix is not bitmap or full
-	GxB_set(A, GxB_SPARSITY_CONTROL, GxB_SPARSE) ;
+	*iter = GB_MALLOC(1, GxB_MatrixTupleIter) ;
+	_MatrixTupleIter_init(*iter, A) ;
+	return (GrB_SUCCESS) ;
```

Comment:
In case `_MatrixTupleIter_init` fails, `iter` is leaked.
return result from `_MatrixTupleIter_init`